caddyhttp/basicauth/basicauth.go

```diff
@@ -63,7 +63,7 @@ func (a BasicAuth) ServeHTTP(w http.ResponseWriter, r *http.Request) (int, error
 
 			// let upstream middleware (e.g. fastcgi and cgi) know about authenticated
 			// user; this replaces the request with a wrapped instance
-			r = r.WithContext(context.WithValue(r.Context(),
+			*r = *r.WithContext(context.WithValue(r.Context(),
```

Comment:
sort of basically what @jamesboswell said.  Simply put if I don't dereference r when calling `withContext` the context value is not available to the replacer.